### PR TITLE
change: fix wrong info `Warning! Running apisix under /root ...`when `make init` at non-root path

### PR DESCRIPTION
**Pull request by totemofwolf**

### CODE REVIEW

**Comment by membphis** [comment]
@totemofwolf what is your current home path?

we can print variable `apisix_home`. I can not understand why you got the warning message.

File: bin/apisix [hunk 1]
```diff
@@ -52,7 +52,7 @@ if script_path:sub(1, 2) == './' then
         error("failed to fetch current path")
     end
 
-    if string.match(apisix_home, '^/[root][^/]+') then
+    if string.match(apisix_home .. "/", '^[/root]*/$') then
```
Comment (membphis) [inline bin/apisix:55]: I do not why the new regex can work correctly :(

Comment (totemofwolf) [reply bin/apisix:55]: > @totemofwolf what is your current home path?
> 
> we can print variable `apisix_home`. I can not understand why you got the warning message.


```

➜  /root1 pwd
/root1


    -- if string.match(apisix_home .. "/", '^[/root]*/$') then
    if string.match(apisix_home, '^/[root][^/]+') then
            is_root_path = true
    end
...
end
print("apisix_home: ", apisix_home)

➜  /root1 make init
./bin/apisix init
apisix_home: 	/root1
Warning! Running apisix under /root is only suitable for development environments and it is dangerous to do so. It is recommended to run APISIX in a directory other than /root. 
./bin/apisix init_etcd
apisix_home: 	/root1

```

----


`if string.match(apisix_home .. "/", '^[/root]*/$') then` works both for `/root1`(no warning) and `/root` (with warning) which is the correct logic ;-)

```

➜  /root1 pwd
/root1

    if string.match(apisix_home .. "/", '^[/root]*/$') then
    -- if string.match(apisix_home, '^/[root][^/]+') then
            is_root_path = true
    end

...
end
print("apisix_home: ", apisix_home)

➜  /root1 make init
./bin/apisix init
apisix_home: 	/root1
./bin/apisix init_etcd
apisix_home: 	/root1

```

Comment (membphis) [reply bin/apisix:55]: Wow, your example is very special

Comment (membphis) [reply bin/apisix:55]: `if string.match(apisix_home .. "/", '^/root/') then` 

You can make a try in this way, this way is better.

Comment (totemofwolf) [reply bin/apisix:55]: > `if string.match(apisix_home .. "/", '^/root/') then`
> 
> You can make a try in this way, this way is better.


```
    if string.match(apisix_home .. "/", '^[/root]*/$') then
    -- if string.match(apisix_home, '^/[root][^/]+') then
    if string.match(apisix_home .. "/", '^/root/') then
            is_root_path = true
    end
```

It works except ` if string.match(apisix_home, '^/[root][^/]+') then`

**Comment by membphis** [comment]
@totemofwolf I think we can close this PR and push a new PR. I don't have the ability to review this PR, it is too big.

image

File: bin/apisix (hunk 1 above)
Comment (membphis) [reply bin/apisix]: I think only `'^/root/'` is right, it means the prefix should be `/root/`.

`^[/root]*/$` is wrong, here is an example:

```shell
$ resty -e 'ngx.say(string.match("/roo/", "^[/root]*/$"))'
/roo/
```

**Comment by totemofwolf** [comment]
> @totemofwolf I think we can close this PR and push a new PR. I don't have the ability to review this PR, it is too big.
> 
> image
> 

OK

**Pull request closed without merging**